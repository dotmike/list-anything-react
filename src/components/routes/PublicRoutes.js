import React from "react";
import PropTypes from "prop-types";
import { Route } from "react-router-dom";

import GuestRoute from "../routes/GuestRoute";

import HomePage from "../pages/HomePage";
import LoginPage from "../pages/LoginPage";
import SignupPage from "../pages/SignupPage";
import ConfirmationPage from "../pages/ConfirmationPage";
import ForgotPasswordPage from "../pages/ForgotPasswordPage";
import ResetPasswordPage from "../pages/ResetPasswordPage";

const PublicRoutes = ({ location }) => (
  <div id="public-routes-container">
    <Route location={location} path="/" exact component={HomePage} />
    <GuestRoute location={location} path="/login" exact component={LoginPage} />
    <GuestRoute
      location={location}
      path="/signup"
      exact
      component={SignupPage}
    />
    <Route
      location={location}
      path="/confirmation/:token"
      exact
      component={ConfirmationPage}
    />
    <GuestRoute
      location={location}
      path="/forgot_password"
      exact
      component={ForgotPasswordPage}
    />
    <GuestRoute
      location={location}
      path="/reset_password/:token"
      exact
      component={ResetPasswordPage}
    />
  </div>
);

PublicRoutes.propTypes = {
  // ownProps
  location: PropTypes.shape({
    pathname: PropTypes.string.isRequired
  }).isRequired
};

export default PublicRoutes;
