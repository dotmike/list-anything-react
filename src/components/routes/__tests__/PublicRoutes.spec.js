import PublicRoutes from "../PublicRoutes";

const setup = (propOverrides = {}, pathname = "/login") => {
  const props = {
    history: {
      push: jest.fn(),
      location: { pathname }
    },
    location: { pathname },
    showContent: false,
    ...propOverrides
  };

  return {
    props,
    wrapperShallow: wrapperShallow(PublicRoutes, props, true)
  };
};

describe("PublicRoutes", () => {
  it("renders correctly when `showContent` is false", () => {
    const { wrapperShallow: wrapper } = setup({ showContent: false });
    expect(wrapper()).toMatchSnapshot();
  });

  it("renders correctly when `showContent` is true", () => {
    const { wrapperShallow: wrapper } = setup({ showContent: true });
    expect(wrapper()).toMatchSnapshot();
  });
});
