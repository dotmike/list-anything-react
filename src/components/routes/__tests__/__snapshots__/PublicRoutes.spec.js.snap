// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PublicRoutes renders correctly when \`showContent\` is false 1`] = `
<div
  id="public-routes-container"
  style={
    Object {
      "display": "none",
    }
  }
>
  <Connect(GuestRoute)
    component={[Function]}
    exact={true}
    location={
      Object {
        "pathname": "/login",
      }
    }
    path="/login"
  />
  <Connect(GuestRoute)
    component={[Function]}
    exact={true}
    location={
      Object {
        "pathname": "/login",
      }
    }
    path="/signup"
  />
  <Connect(GuestRoute)
    component={[Function]}
    exact={true}
    location={
      Object {
        "pathname": "/login",
      }
    }
    path="/forgot_password"
  />
  <Connect(GuestRoute)
    component={[Function]}
    exact={true}
    location={
      Object {
        "pathname": "/login",
      }
    }
    path="/reset_password/:token"
  />
</div>
`;

exports[`PublicRoutes renders correctly when \`showContent\` is true 1`] = `
<div
  id="public-routes-container"
  style={
    Object {
      "display": "block",
    }
  }
>
  <Connect(GuestRoute)
    component={[Function]}
    exact={true}
    location={
      Object {
        "pathname": "/login",
      }
    }
    path="/login"
  />
  <Connect(GuestRoute)
    component={[Function]}
    exact={true}
    location={
      Object {
        "pathname": "/login",
      }
    }
    path="/signup"
  />
  <Connect(GuestRoute)
    component={[Function]}
    exact={true}
    location={
      Object {
        "pathname": "/login",
      }
    }
    path="/forgot_password"
  />
  <Connect(GuestRoute)
    component={[Function]}
    exact={true}
    location={
      Object {
        "pathname": "/login",
      }
    }
    path="/reset_password/:token"
  />
</div>
`;
